src/packet/bwe/arrival_group.rs

```diff
@@ -51,38 +54,47 @@ impl ArrivalGroup {
             return Belongs::Yes;
         };
 
-        let Some(send_diff) = packet
+        let Some(first_send_delta) = packet
             .local_send_time
             .checked_duration_since(first_local_send_time)
         else {
             // Out of order
             return Belongs::Skipped;
         };
 
-        if send_diff < BURST_TIME_INTERVAL {
-            // Sent within the same burst interval
-            return Belongs::Yes;
-        }
+        let inter_arrival_time = {
+            let last_remote_recv_time = self.remote_recv_time();
 
-        let inter_arrival_time = packet
-            .remote_recv_time
-            .checked_duration_since(self.remote_recv_time());
+            if packet.remote_recv_time >= last_remote_recv_time {
+                (packet.remote_recv_time - last_remote_recv_time).as_secs_f64()
+            } else {
+                (last_remote_recv_time - packet.remote_recv_time).as_secs_f64() * -1.0
```

Comment:
Is the `* -1.0` the reason we convert to floats? I might not have the full picture so take with a grain of salt but it could potentially be modeled more type-safe if we do:

```rust
enum SignedDuration {
	Positive(Duration),
	Negative(Duration)
}
```

Reply:
I was confused by this code. Until I remembered that `Duration` cannot be negative :( I think having a wrapper type like this would be good, but probably a good future improvement

Reply:
In ureq I did this: https://github.com/algesten/ureq/blob/main/src/unversioned/transport/time.rs#L9-L24

I want to copy this code across to str0m and merge it and replace our similar concepts here: https://github.com/algesten/str0m/blob/main/src/util/time_tricks.rs#L6-L10

This new wrapper type could also handle negative durations.

Reply:
> Is the * -1.0 the reason we convert to floats?

Well it [was](https://github.com/algesten/str0m/blob/b6905402fff3e23b7ea3d1a85ace7bdedad000c7/src/packet/bwe/arrival_group.rs#L76-L77) being converted anyway, but for the pretty much the same reason: so `inter_group_delay_delta` could be negative.

But yeah, difference in time calculations is a problem: I can imagine that there might be other cases when `Instant`s subtraction returns `Duration::ZERO` when in the libwebrtc's implementation that would be a negative [`TimeDelta`](https://source.chromium.org/chromium/chromium/src/+/main:third_party/webrtc/api/units/time_delta.h;drc=e0a524b5e02f293a0c13fd687334f7e6eedb971d).

Reply:
@alexlapa sounds great!

You can join our Zulip if you want a simpler way to chat ;)